File: ch20/q1/plugins/words1.rb
def extract_words(path_to_file)
  word_list = IO.read(path_to_file).gsub(/[\W_]+/, ' ').downcase.split
  stop_words = IO.read('../../stop_words.txt').split(',') + ('a'..'z').to_a
  word_list.reject{|w| stop_words.include?(w)}
end

File: ch20/q6_1/plugins/words1.py
import re

def read_words(path_to_file):
    with open('../../stop_words.txt') as f:
        stop_words = f.read().split(',')
    
    words = []
    with open(path_to_file) as f:
        for i, line in enumerate(f):
            for word in re.findall(r'[a-z]{2,}', line.lower()):
                if word not in stop_words:
                    words.append((i//45+1, word))
    return words